setup.py

```diff
@@ -39,28 +35,38 @@ def find_libxml2_include():
             include_dirs.append(d)
     return include_dirs
 
+
+class CustomBuildExt(build_ext):
+    def finalize_options(self):
+        build_ext.finalize_options(self)
+
+        __builtins__.__NUMPY_SETUP__ = False
+
+        import Cython
+        import lxml
+        import numpy
+
+        include_dirs = [numpy.get_include()] + lxml.get_include() + find_libxml2_include()
+
+        self.include_dirs.extend(include_dirs)
+
 ext_modules = [
     Extension('dragnet.lcs',
-              sources=["dragnet/lcs.pyx"],
-              include_dirs=[get_include()],
+              sources=["dragnet/lcs.cpp"],
```

Comment:
How does this work without checking in the generated C++ files?

Reply:
I am not sure what you mean. Writing this is in fact unrolling what the cythonize function does under the hood with source names etc.

Reply:
Yeah, but I didn't see any call to cythonize so I want sure if it was still generating those files or if those files needed to be checked in to be found.

I've seen other projects just check in the generated files to avoid using cythonize during setup.py. The downside is you have to remember to update those before checking in and it makes the diffs larger.

Reply:
Oh. I see now. I did not understand "checking" this way, hence the confusion.

Reply:
When I run:
```
pip install git+https://github.com/Yomguithereal/dragnet.git@setup-build-enhancements
```
The command fails, and I see the following error:
```
  clang: error: no such file or directory: 'dragnet/lcs.cpp'
  clang: error: no input files
  error: command 'clang' failed with exit status 1
```

This means that the .cpp files need to be generated before the Extension commands can be called. I'm not opposed to generating the .cpp files, but then we need to ensure they are regenerated after any changes to the .pyx files.